libcomxml/core/__init__.py

```diff
@@ -125,14 +128,16 @@ def element(self, parent=None):
 
         :param parent: an etree Element to be used as parent for this one
         """
+        if self.namespace:
+            name = '{%s}%s' % (self.namespace, self.name)
+        else:
+            name = self.name
         if parent is not None:
-            if self.namespace:
-                name = '{%s}%s' % (self.namespace, self.name)
-            else:
-                name = self.name
-            ele = etree.SubElement(parent, name, **self.attributes)
+            ele = etree.SubElement(parent, name, nsmap=self.nsmap,
+                                   **self.attributes)
         else:
-            ele = etree.Element(self.name, **self.attributes)
+            ele = etree.Element(name, nsmap=self.nsmap,
+                                **self.attributes)
```

Comment:
The `nsmap` parameter could be included in `attributes` because they are forwarded to `etree.Element`, but this is more clear and the old way will still working.

```python
self.tag = XmlField('feed', attributes={'nsmap': {
    'opensearch': 'http://a9.com/-/spec/opensearch/1.1/'
}})
```